pkg/cloudprovider/vsphereparavirtual/cloud.go

```diff
@@ -112,12 +112,12 @@ func (cp *VSphereParavirtual) Initialize(clientBuilder cloudprovider.ControllerC
 	klog.V(0).Info("Initing vSphere Paravirtual Cloud Provider")
 
 	if vpcModeEnabled {
-		if podIPPoolType != PublicIPPoolType && podIPPoolType != PrivateIPPoolType {
-			klog.Fatalf("Pod IP Pool Type can be either Public or Private in VPC network, %s is not supported", podIPPoolType)
+		if podIPPoolType == "" {
+			klog.Fatalf("Pod IP Pool Type can't be empty in VPC network")
 		}
 
-		if podIPPoolType == "" {
-			podIPPoolType = PrivateIPPoolType
+		if podIPPoolType != PublicIPPoolType && podIPPoolType != PrivateIPPoolType {
+			klog.Fatalf("Pod IP Pool Type can be either Public or Private in VPC network, %s is not supported", podIPPoolType)
```

Comment:
msg: --pod-ip-pool-type can be either Public or Private in the NSX-T VPC network, %s is not supported